### DISCUSSION TITLE
ActualWidth and ActualHeight always return 0 and 1 respectively.

### DISCUSSION THREAD

**Discussion by onslauth** [microsoft/WindowsAppSDK#2150] [Q&A] [answered]
When trying to obtain a controls ActualWidth and ActualHeight, the returned values are always 0 and 1. 

I have added event listeners for multiple controls' 'Loaded' events, and each one returns a width and height of 0 and 1. I have even included a PointerWheelChanged event listener, so that I could be sure that the controls have rendered before trying to obtain their ActualWidth and ActualHeight values, and the result is the same, width and height are always 0 and 1.

There is a demo application at: [https://github.com/onslauth/CanvasPanTest](https://github.com/onslauth/CanvasPanTest).

VisualStudio 2022
AppSDK 1.0
WinUI 3.0 Desktop Application

Does anyone have any suggestions as to the correct way to resolve the problem and get the ActualWidth and ActualHeight values?

**Comment by castorix** [comment, marked as answer]
Fix Debug.WriteLine, like :

`System.Diagnostics.Debug.WriteLine(String.Format("Width : {0} Height : {1}", width, height));`

and I get :
`Width : 6016 Height : 4016`

**Reply by onslauth** [reply]
I cannot believe that was the issue. Thank you so much for the help.
